bluecellulab/tools.py

```diff
@@ -634,64 +634,102 @@ def compute_memodel_properties(
     return {"holding_current": holding_current, "rheobase": rheobase, "rin": rin}
 
 
-def list_segment_variables(cell, xs=(0.1, 0.5, 0.9)):
-    """{section_name: {x: [ 'v', 'ina', 'ik', 'ena', 'gna', ...
-
-    ]}}
+def list_segment_ion_variables(cell, xs=(0.5,)):
+    """Enumerate top-level ionic variables (e.g. v, ina, ik, gna) available for
+    each section at specified segment locations.
+
+    Includes:
+      - Membrane potential (v)
+      - Standard ionic currents (ina, ik, ica, icl)
+      - Reversal potentials (ena, ek, eca, ecl)
+      - Conductances if exposed (gna, gk, gca, gcl)
     """
     out = {}
-    prefix = cell.soma.name().split('.')[0]
-    secs = [sec for sec in neuron.h.allsec() if sec.name().startswith(prefix)]
-
-    for sec in secs:
+    for sec in cell.sections.values():
         xmap = {}
         for x in xs:
             seg = sec(x)
-            names = set(a[5:] for a in dir(seg) if a.startswith("_ref_"))
+
+            vars_ = []
+            # voltage always there
             if hasattr(seg, "_ref_v"):
-                names.add("v")
-            xmap[float(x)] = sorted(names)
+                vars_.append("v")
+
+            # ionic currents (present if mechanisms with ions are inserted)
+            for ion in ("na", "k", "ca", "cl"):
```

Comment:
Does that cover all the cases? what about non-spepcific currents?

@darshanmandge maybe you have on opinion on this?

Reply:
I think it may not. We have `NONSPECIFIC _CURRENT`s, `i_pas` and some use defined ionic currents. I think @ilkilic can use the logic used in bluepyemodel to get [all currents and concentrations](https://github.com/openbraininstitute/BluePyEModel/blob/71f8a58aab1fa0a9a3c9c5c5d70e099b5f5a7f18/bluepyemodel/access_point/access_point.py#L277) if it is possible to do implement something similar here. What do you think?